### ISSUE TITLE
Be more precise in the URL shortener section of the Out Of Band Protocol

### ISSUE THREAD

**Issue by swcurran** [hyperledger/aries-rfcs#470]
The definition of the [URL Shortener](https://github.com/hyperledger/aries-rfcs/tree/master/features/0434-outofband#url-shortening) in the Out of Band (OOB) protocol was based on experience with the same feature in the [vc-authn-oidc](https://github.com/bcgov/vc-authn-oidc) repo. An example of a shortened URL from vc-authn-oidc can (currently) be generated [here](https://lngpipeline-camp.vonx.io/authorize). Note that the vc-authn-oidc implementation does not exactly match the OOB RFC in the format of the shortened URL or the name of the query parameter in the expanded URL.  Please ignore those details. The important issue here is how the server handles the resolution of the GET of the shortened URL.

Several vendors have implemented handling for the vc-authn-oidc approach, but another raised this concern:

```
As per our testing we found that shortened url is not working in intended way.
When I say intended way, I mean following steps:

1. Send GET request to shortened URL
2. Shortened URL redirect to full URL
3. Sender of GET request follow redirect to full URL
4. Full URL returns actual data

As of now above steps are not happening. Instead following is what is happening:

1. Send GET request to encoded URL
2. Ignore response of GET request, instead take location header of the response
3. Do not follow the redirect given by the location header. Instead, parse location header URL, and take value of query string param "m"
4. Base64 decode this value of "m" and get presentation request

So, there is disconnect at steps 2, 3 and 4.
```

We would like to be sure that this is clarified in the OOB RFC.

How can we improve the URL shortener section of the OOB RFC to make sure that we don't have any confusion in implementing it across agents?

**Comment by swcurran** [comment]
A first note on this.

The [stated intention](https://github.com/hyperledger/aries-rfcs/tree/master/features/0434-outofband#standard-out-of-band-message-encoding) of the full URL in the OOB RFC is that if the person is resolving the URL in a browser, the query parameter is to be held on to, a page displayed with links to how to get an agent, and the query parameter passed via the "get agent" links in hopes that after the user installs their agent, the fully provisioned agent will have the query parameter available and will interpret it and seemlessly continue with the flow started by the OOB message.

Is it a reasonable expectation that such deep-link processing is going to happen, or should the provisioning of the agent be out of scope of the OOB process?

I don't know the actual answer to that, but it would surprise me that such the "user has no agent" flow would work as described.  In all of the use cases we have implemented, we've instructed users to get their Mobile Agent before they get to the point of an OOB message, and so assume they will use the scanner in the mobile agent to read the OOB QR code message, or that they would come back to the OOB message point after they provision their agent.
